### ISSUE TITLE
Recording Twitch chat?

### ISSUE THREAD

**Issue by alystair** [Franiac/TwitchLeecher#488]
I'm curious if there'd be a way to encode Twitch chat in tandem with clips, would probably be a PITA to code...

**Comment by Franiac** [comment]
I don't know if the new API has the capability, but V5 (the API TL is currently using) does not provide any way to download the chat as far as I know.

**Comment by fgl27** [comment]
the link for vod chat is

https://api.twitch.tv/v5/videos/vod.id/comments?client_id=$yours_app_client_id&content_offset_seconds=$vod.offset

vod.id and vod.offset is the info from the result of the clip api, if the clip doesn't have a vod info (aka the vod property is null) the chat is deleted

but anyway embed that on a clip, that is a lot of work on my view.


vod api clip sample to put a context

```
{
	"slug": "FragileCooperativeMochaBabyRage",
	"tracking_id": "233816142",
	"url": "https://clips.twitch.tv/FragileCooperativeMochaBabyRage?tt_medium=clips_api\u0026tt_content=url",
	"embed_url": "https://clips.twitch.tv/embed?clip=FragileCooperativeMochaBabyRage\u0026tt_medium=clips_api\u0026tt_content=embed",
	"embed_html": "\u003ciframe src='https://clips.twitch.tv/embed?clip=FragileCooperativeMochaBabyRage\u0026tt_medium=clips_api\u0026tt_content=embed' width='640' height='360' frameborder='0' scrolling='no' allowfullscreen='true'\u003e\u003c/iframe\u003e",
	"broadcaster": {
		"id": "146029678",
		"name": "girlwithyellowspoon",
		"display_name": "girlwithyellowspoon",
		"channel_url": "https://www.twitch.tv/girlwithyellowspoon",
		"logo": "https://static-cdn.jtvnw.net/jtv_user_pictures/45efaa79-f45e-444e-94c5-e0e1dd2bc472-profile_image-150x150.png"
	},
	"curator": {
		"id": "42400114",
		"name": "dev0ti0n",
		"display_name": "dev0ti0n",
		"channel_url": "https://www.twitch.tv/dev0ti0n",
		"logo": "https://static-cdn.jtvnw.net/user-default-pictures/27103734-3cda-44d6-a384-f2ab71e4bb85-profile_image-150x150.jpg"
	},
	"vod": {
		"id": "256787513",
		"url": "https://www.twitch.tv/videos/256787513?t=2h27m45s",
		"offset": 8865,
		"preview_image_url": "https://static-cdn.jtvnw.net/s3_vods/9c4b97d75a4e1029c971_girlwithyellowspoon_28545932448_852790942/thumb/thumb0-320x240.jpg"
	},
	"broadcast_id": "28545932448",
	"game": "IRL",
	"language": "en",
	"title": "Not having DSLs",
	"views": 457045,
	"duration": 13.33,
	"created_at": "2018-05-02T05:31:46Z",
	"thumbnails": {
		"medium": "https://clips-media-assets2.twitch.tv/233816142-preview-480x272.jpg",
		"small": "https://clips-media-assets2.twitch.tv/233816142-preview-260x147.jpg",
		"tiny": "https://clips-media-assets2.twitch.tv/233816142-preview-86x45.jpg"
	}
}
```

null vod

```
{
	"slug": "ElatedSwissSquidPeteZaroll",
	"tracking_id": "232251774",
	"url": "https://clips.twitch.tv/ElatedSwissSquidPeteZaroll?tt_medium=clips_api\u0026tt_content=url",
	"embed_url": "https://clips.twitch.tv/embed?clip=ElatedSwissSquidPeteZaroll\u0026tt_medium=clips_api\u0026tt_content=embed",
	"embed_html": "\u003ciframe src='https://clips.twitch.tv/embed?clip=ElatedSwissSquidPeteZaroll\u0026tt_medium=clips_api\u0026tt_content=embed' width='640' height='360' frameborder='0' scrolling='no' allowfullscreen='true'\u003e\u003c/iframe\u003e",
	"broadcaster": {
		"id": "11249217",
		"name": "jakenbakelive",
		"display_name": "JakenbakeLIVE",
		"channel_url": "https://www.twitch.tv/jakenbakelive",
		"logo": "https://static-cdn.jtvnw.net/jtv_user_pictures/jakenbakelive-profile_image-cf0960d3d360b6c9-150x150.png"
	},
	"curator": {
		"id": "48208808",
		"name": "just_joshing",
		"display_name": "Just_Joshing",
		"channel_url": "https://www.twitch.tv/just_joshing",
		"logo": "https://static-cdn.jtvnw.net/jtv_user_pictures/just_joshing-profile_image-9a45acab989899f6-150x150.jpeg"
	},
	"vod": null,
	"broadcast_id": "28508497024",
	"game": "IRL",
	"language": "en",
	"title": "Jake Responds to \"King of IRL\"",
	"views": 79216,
	"duration": 43,
	"created_at": "2018-04-29T08:55:08Z",
	"thumbnails": {
		"medium": "https://clips-media-assets2.twitch.tv/232251774-preview-480x272.jpg",
		"small": "https://clips-media-assets2.twitch.tv/232251774-preview-260x147.jpg",
		"tiny": "https://clips-media-assets2.twitch.tv/232251774-preview-86x45.jpg"
	}
}
```

**Comment by Franiac** [comment]
The comments API is not official. There is no documentation in the official reference. It's the same like the sub-only API, it is there for Twitch to use but not for external devs. This means it can be changed whenever Twitch like to do so. This would break TL everytime they do.

The API examples you posted are clips. Clips are not the same as videos. Not that it makes any difference, just pointing that out.

**Comment by fgl27** [comment]
You are confusing a lot.

Undocumented doesn't means forbidden after all why a third party client id works, if they wanna to forbid something they will even if is a documented api and for no reason.

twitch can change any api documented or not at any time without any reason without any consultation from how uses they api, as they have be doing for ever, so any one that makes a app that depends on any documented or not api must be ready as things on twitch side are very confusing as how they work.

> The API examples you posted are clips. Clips are not the same as videos. Not that it makes any difference, just pointing that out.

what you are talking about ? did you read the first comment?

The first comment has this:

> I'm curious if there'd be a way to encode Twitch chat in tandem with **clips**

I inform exactly how to get the position on vod for the particularly chat of a clip, from the result of the clip api you get the info, that is why I share toose samples... that show where `vod.id` and `vod.offset` are coming from and when it is `null` and it's meaning, perfect answer on how to get the clip information requested by the first comment, whether one can "encode" that on a clip that is another thing.

My answer is perfect clear I was just trying to show you that is possible to get the chat info not just for a vod but for a specific clip from a vod, if you don't wan't to use is not particularly my problem, I was just inform you the how to.

**Comment by Franiac** [comment]
@fgl27 You can step down from your throne now, nobody hurt your feelings...

The first comment is confusing in itself since Twitch Leecher does not support downloading clips in the first place, so the issue is obsolete anyway. We are discussing a new feature (chat) for a feature (clips) that is not there.... duh.

Using unofficial API's lead TL to drop sub-only download support in 1.8, so I will learn from that mistake and won't intruduce any new features using another undocumented private API. Twitch devs clearly warn from using private API's everywhere in the forums. If you don't care, ok, but I care. Tried it once, fell flat on my face, won't try it a second time, it's as easy as that.

Public API's don't change with the exception of Twitch messing stuff up at times. But deliberate changes are made in new API versions because that's what they are for.

#174 Yes I know how the clips API works because there is a whole issue for that...

Long story short: When I have time to spend on the project, downloading clips will be poissible in the future, downloading chat won't, as long as the API is private.

**Comment by fgl27** [comment]
> @fgl27 You can step down from your throne now, nobody hurt your feelings...

Again what you are talking about? rhetorical BTW.

Someone try to help you, you didn't care then you go and disrespect the person for no reason, makes no sense, you wanna talk with someone do, but spare the unnecessary comments as the one you just used, after all I was just being serious on my replies assuming you didn't fully understand my point, I don't see disrespects on my end so yours phrasing makes no sense.

Don't you see I was trying to help you? again rhetorical.

> #174 Yes I know how the clips API works because there is a whole issue for that...

But when you look to a commit that is there for years no solution one can easily put 1 and 2 together, no one helps you because when they try, they get dismissed or disrespect for no reason.

By, I'll never come here again ever, don't even bother with yours odd disrespectful replies spare yourself my lord.

**Comment by Franiac** [comment]
> The comments API is not official. There is no documentation in the official reference. It's the same like the sub-only API, it is there for Twitch to use but not for external devs. This means it can be changed whenever Twitch like to do so. This would break TL everytime they do.
The API examples you posted are clips. Clips are not the same as videos. Not that it makes any difference, just pointing that out.

This was my answer to your comment. I don't see any shots fired but you follow up with aggressive phrasing. Have a good one, I won't miss you here...
